vault/resource_quota_rate_limit.go

```diff
@@ -43,6 +45,21 @@ func quotaRateLimitResource() *schema.Resource {
 				Description:  "The maximum number of requests at any given second to be allowed by the quota rule. The rate must be positive.",
 				ValidateFunc: validation.FloatAtLeast(0.0),
 			},
+			"interval": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The duration to enforce rate limiting for (default \"1s\").",
+				StateFunc:    durationSecond,
+				ValidateFunc: validateDurationSecond,
+				Default:      "1s",
+			},
+			"block_interval": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "If set, when a client reaches a rate limit threshold, the client will be prohibited from any further requests until after the 'block_interval' has elapsed.",
+				StateFunc:    durationSecond,
```

Comment:
Because read requests return these values in seconds and not a duration the configuration value must be computed in terms of seconds to prevent a perpetual diff.

Reply:
Since vault converts the duration strings to seconds (int), we prefer to only support `schema.TypeInt` for these sorts of Vault fields. This should greatly simplify your PR.